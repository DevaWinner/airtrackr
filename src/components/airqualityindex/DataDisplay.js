import React from 'react';
import PropTypes from 'prop-types';

const DataDisplay = ({ name, value }) => (
  <div className="detail-data flex">
    <p className="data-name">
      <div dangerouslySetInnerHTML={{ __html: name }} />
    </p>
    <p className="data-value">
      {value}
    </p>
  </div>
);

DataDisplay.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

export default DataDisplay;
