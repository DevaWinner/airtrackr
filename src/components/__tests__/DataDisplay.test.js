import React from 'react';
import { render } from '@testing-library/react';
import DataDisplay from '../airqualityindex/DataDisplay';

describe('DataDisplay', () => {
  it('should render correctly with props', () => {
    const dataname = 'NO';
    const data = '2';
    const { container } = render(<DataDisplay dataname={dataname} data={data} />);

    expect(container).toMatchSnapshot();
  });
});
