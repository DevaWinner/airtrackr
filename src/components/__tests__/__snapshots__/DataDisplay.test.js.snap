// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataDisplay should render DataDisplay component 1`] = `[Function]`;

exports[`DataDisplay should render correctly with props 1`] = `
<div>
  <div
    class="detail-data flex"
  >
    <p
      class="data-name"
    >
      <div />
    </p>
    <p
      class="data-value"
    />
  </div>
</div>
`;
